NXP/MCIMX8M_EVK_4GB/Library/DfciDeviceIdSupportLib/Imx8DeivceIdSupportLib.c

```diff
@@ -0,0 +1,162 @@
+/** @file
+DfciDeviceidSupportLib.c
+
+This library provides access to platform data the becomes the DFCI DeviceId.
+
+Copyright (c) 2019, Microsoft Corporation
+
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+**/
+
+#include <PiDxe.h>
+#include <Library/DfciDeviceIdSupportLib.h>
+#include <Library/BaseLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+
+/**
+Gets the serial number for this device.
+
+@para[out] SerialNumber - UINTN value of serial number
+
+@return EFI_SUCCESS - SerialNumber has been updated to equal the serial number of the device
+@return EFI_ERROR   - Error getting number
+
+**/
+EFI_STATUS
+EFIAPI
+DfciIdSupportV1GetSerialNumber(
```

Comment:
These (product name, serial number, etc.) should be pulled out of the SMBIOS structures and then should be made part of the Silicon/IMX generic library.  Or vice-versa, the SMBIOS code should pull this information from here.

Reply:
Some background info to think about: We have some ongoing discussions here about how to allow an OEM to update some of the [SMBIOS data used for CHID ](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/specifying-hardware-ids-for-a-computer). Mainly focused on things like SKU, Product Name, etc. The request from manufacturers is to be able to set this at manufacture time on the line, rather than compile it in. We don't have a concrete plan on how to do so yet, but we are thinking about using the UEFI variable store.

Reply:
For now this is just a fake/stub.  If/when SMBIOS is figured out this can be updated.